app/models/concerns/gias_helpers.rb

```diff
@@ -0,0 +1,20 @@
+module GIASHelpers
+  extend ActiveSupport::Concern
+
+  included do
+    scope :in_gias_schools, -> { includes(:gias_school).references(:gias_schools) }
+
+    scope :currently_open, -> { in_gias_schools.where(gias_school: { status: GIAS::Types::OPEN_STATUSES }) }
+    scope :eligible_establishment_type, -> { in_gias_schools.where(gias_school: { type_name: GIAS::Types::ELIGIBLE_TYPES }) }
+    scope :in_england, -> { in_gias_schools.where(gias_school: { in_england: true }) }
+    scope :section_41, -> { in_gias_schools.where(gias_school: { section_41_approved: true }) }
+    scope :eligible, -> { currently_open.eligible_establishment_type.in_england.or(currently_open.in_england.section_41) }
+    scope :cip_only, -> { currently_open.where(gias_school: { type_name: GIAS::Types::CIP_ONLY_TYPES }) }
+    scope :eligible_or_cip_only, -> { eligible.or(cip_only) }
+    scope :not_cip_only, -> { where.not(id: cip_only) }
```

Comment:
If you look in [GIAS::SchoolRow](https://github.com/DFE-Digital/register-early-career-teachers-public/blob/main/app/services/gias/school_row.rb) we do lots of this eligibility checking up front when we import/update schools from GIAS.

We should be able to use that (it might need to be tweaked/extended) but it'll save having to use complex logic/queries.

Reply:
If for any reason we can't use it as-is, we should either update what we have or add to it so we can apply the logic once at import and then use it via a single scope.